src/software/backend/simulation/physics/physics_robot_test.cpp

```diff
@@ -84,228 +305,551 @@ TEST(PhysicsRobotTest, test_physics_robot_dimensions_left_side_outside_radius)
+        physics_robot.applyWheelForceFrontRight(-0.5);
+
         // 5 and 8 here are somewhat arbitrary values for the velocity and position
         // iterations but are the recommended defaults from
         // https://www.iforce2d.net/b2dtut/worlds
         world->Step(1.0 / 60.0, 5, 8);
     }
 
-    auto ball = physics_ball.getBallWithTimestamp(Timestamp::fromSeconds(0));
-    EXPECT_GT(ball.velocity().orientation().minDiff(Angle::half()).toDegrees(), 0.1);
+    auto robot = physics_robot.getRobotWithTimestamp(Timestamp::fromSeconds(0));
+    EXPECT_LT(robot.velocity().x(), -1.0);
+    EXPECT_NEAR(robot.velocity().y(), 0, 1e-5);
+    EXPECT_NEAR(robot.angularVelocity().toDegrees(), 0, 1);
+
+    EXPECT_LT(robot.position().x(), -0.5);
+    EXPECT_NEAR(robot.position().y(), 0, 1e-5);
+    EXPECT_NEAR(robot.orientation().toDegrees(), 0, 1);
 }
 
-// Roll the ball along the back side of the robot just outside of the robot radius
-// and check it does not collide
-TEST(PhysicsRobotTest, test_physics_robot_dimensions_back_side_outside_radius)
+TEST_F(PhysicsRobotTest, test_robot_spin_clockwise)
 {
     b2Vec2 gravity(0, 0);
     auto world = std::make_shared<b2World>(gravity);
 
     Robot robot_parameter(0, Point(0, 0), Vector(0, 0), Angle::zero(),
                           AngularVelocity::zero(), Timestamp::fromSeconds(0));
-    PhysicsRobot physics_robot(world, robot_parameter);
-
-    // The ball should pass right next to the robot without colliding, so should not
-    // change direction
-    Ball ball_parameter(Point(-ROBOT_MAX_RADIUS_METERS - BALL_MAX_RADIUS_METERS, 1),
-                        Vector(0, -2), Timestamp::fromSeconds(0));
-    auto physics_ball = PhysicsBall(world, ball_parameter);
+    PhysicsRobot physics_robot(world, robot_parameter, 1.0);
 
     // We have to take lots of small steps because a significant amount of accuracy
     // is lost if we take a single step of 1 second
     for (unsigned int i = 0; i < 60; i++)
     {
+        physics_robot.applyWheelForceFrontLeft(-0.5);
+        physics_robot.applyWheelForceBackLeft(-0.5);
+        physics_robot.applyWheelForceBackRight(-0.5);
+        physics_robot.applyWheelForceFrontRight(-0.5);
+
         // 5 and 8 here are somewhat arbitrary values for the velocity and position
         // iterations but are the recommended defaults from
         // https://www.iforce2d.net/b2dtut/worlds
         world->Step(1.0 / 60.0, 5, 8);
     }
 
-    auto ball = physics_ball.getBallWithTimestamp(Timestamp::fromSeconds(0));
-    EXPECT_LT(ball.velocity().orientation().minDiff(Angle::threeQuarter()).toDegrees(),
-              0.1);
+    auto robot = physics_robot.getRobotWithTimestamp(Timestamp::fromSeconds(0));
+    // Because the robot's center of mass is not perfect we expect it to have drifted a
+    // little bit while spinning
+    EXPECT_LT((robot.position() - Point(0, 0)).length(), 0.05);
+    EXPECT_LT((robot.velocity() - Vector(0, 0)).length(), 0.05);
```

Comment:
same as below

Reply:
Both are intentionally redundant for explicitness, so they match the initial state of the robot specified at the start of the test